src/Common/SystemLogBase.cpp

```diff
@@ -31,30 +31,25 @@ namespace ErrorCodes
     extern const int TIMEOUT_EXCEEDED;
 }
 
-namespace
-{
-    constexpr size_t DBMS_SYSTEM_LOG_QUEUE_SIZE = 1048576;
-}
-
 ISystemLog::~ISystemLog() = default;
 
 
 template <typename LogElement>
-SystemLogQueue<LogElement>::SystemLogQueue(
-    const String & table_name_,
-    size_t flush_interval_milliseconds_,
-    bool turn_off_logger_)
-    : log(&Poco::Logger::get("SystemLogQueue (" + table_name_ + ")"))
-    , flush_interval_milliseconds(flush_interval_milliseconds_)
+SystemLogQueue<LogElement>::SystemLogQueue(const SystemLogQueueSettings & settings_)
+    : log(&Poco::Logger::get("SystemLogQueue (" + settings_.database + "." +settings_.table + ")"))
+    , settings(settings_)
+
 {
-    if (turn_off_logger_)
+    queue.reserve(settings.reserved_size);
+
+    if (settings.turn_off_logger)
         log->setLevel(0);
 }
 
 static thread_local bool recursive_push_call = false;
 
 template <typename LogElement>
-void SystemLogQueue<LogElement>::push(const LogElement & element)
+void SystemLogQueue<LogElement>::push(LogElement element)
```

Comment:
```suggestion
void SystemLogQueue<LogElement>::push(LogElement && element)
```
(just to be sure it will not copy)

Reply:
Done!